service-tests/helpers/validators.js

```diff
@@ -37,6 +37,8 @@ const isStarRating = withRegex(/^(?=.{5}$)(\u2605{0,5}[\u00BC\u00BD\u00BE]?\u260
 // Required to be > 0, beacuse accepting zero masks many problems.
 const isMetric = withRegex(/^[1-9][0-9]*[kMGTPEZY]?$/);
 
+const isMetricOpenIssues = withRegex(/^[0-9]+[kMGTPEZY]? open$/);
```

Comment:
Could we make sure this is nonzero, as with `isMetric`? We've had broken code which displayed 0 on a malformed response.

Reply:
Alternatively we could add a test with API mock which will allow expect specific value in result.

Reply:
It is legitimate for this to be zero if there are actually zero open issues though. In the GitHub tests (on master), you've got a mix of tests that are looking for `[0-9]+...` (e.g: https://github.com/badges/shields/blob/master/service-tests/github.js#L69-L74 ) and others looking for `[1-9][0-9]+...` (e.g: https://github.com/badges/shields/blob/master/service-tests/github.js#L80-L85 ).

Do you think we should extract both regexes and add tests for both cases, or should we always validate a non-zero value?

Reply:
True, though as long as we pick a project that is likely to stay around, there will always be open issues. I'm always fine with additional tests using mocks if the integrated tests don't seem satisfactory.

I think using `/^[1-9][0-9]*[kMGTPEZY]? open$/` would be fine for both open issues and open issues by label. Does that answer your question?